### PR TITLE
Sort license dropdown

### PR DESCRIPTION
**Pull request by benedikt-richter**
### Summary of changes

* Sort licenses in license dropdown alphabetically.
* For performance reasons `localCompare` is replaced everywhere.

### Context and reason for change

Licenses in license dropdown were not sorted alphabetically.

### How can the changes be tested

Check that in UI packages etc are still sorted correctly.

### CODE REVIEW

File: src/Frontend/Components/PackagePanel/package-panel-helpers.ts [hunk 1]
```diff
@@ -61,9 +61,10 @@ function sortSources(
         attributionSources[sourceA]?.priority -
         attributionSources[sourceB]?.priority
       ) ||
-      attributionSources[sourceA]?.name.localeCompare(
```
Comment (antonbauhofer) [inline src/Frontend/Components/PackagePanel/package-panel-helpers.ts:64]: Why do you prefer your implementation to `localeCompare`?

File: src/Frontend/Components/AttributionColumn/__tests__/LicenseField.test.tsx [hunk 2]
```diff
@@ -58,4 +58,36 @@ describe('The LicenseField', () => {
       'GPL - General Public License',
     ]);
   });
+
+  it('sorts entries', () => {
```
Comment (antonbauhofer) [inline src/Frontend/Components/AttributionColumn/__tests__/LicenseField.test.tsx:62]: We could have a few more entries to sort and include some more exotic names (i.e. containing numbers / umlaute)

File: src/Frontend/util/get-alphabetical-comparer.ts [hunk 3]
```diff
@@ -41,9 +41,9 @@ export function getAlphabeticalComparer(attributions: Attributions) {
     if (elementTitleIsAlphabetical && !otherElementTitleIsAlphabetical)
       return -1;
 
-    return elementTitle.localeCompare(otherElementTitle, undefined, {
-      sensitivity: 'base',
```
Comment (antonbauhofer) [inline src/Frontend/util/get-alphabetical-comparer.ts:45]: what is `sensitivity: 'base'`? Please make sure it's ok to drop that.

**Review by antonbauhofer — APPROVED** [review]
Looks good and works as expected.
Just consider my (optional) comments.

File: src/Frontend/Components/AttributionColumn/LicenseField.tsx [hunk 4]
```diff
@@ -42,6 +42,9 @@ interface LicenseFieldProps extends InputElementProps {
 export function LicenseField(props: LicenseFieldProps): ReactElement {
   const formattedLicenseNamesToShortNameMapping =
     getFormattedLicenseNamesToShortNameMapping(props.frequentLicenseNames);
+  const sortedLicenses = Object.keys(
+    formattedLicenseNamesToShortNameMapping
+  ).sort((a, b) => (a.toLowerCase() < b.toLowerCase() ? -1 : 1));
```
Comment (lennartclaas) [inline src/Frontend/Components/AttributionColumn/LicenseField.tsx:47]: One could think about using more meaningful names than `a` and `b` as long as these do not reduce readability.

File: src/Frontend/Components/PackagePanel/package-panel-helpers.ts (hunk 1 above)
Comment (benedikt-richter) [reply src/Frontend/Components/PackagePanel/package-panel-helpers.ts:64]: The performance of `localCompare` is much lower than with the current implementation.

File: src/Frontend/util/get-alphabetical-comparer.ts (hunk 3 above)
Comment (benedikt-richter) [reply src/Frontend/util/get-alphabetical-comparer.ts:45]:
```
// a negative value: in German, ä sorts before z
'ä'.localeCompare('z', 'de')                               // -1// a positive value: in Swedish, ä sorts after z
'ä'.localeCompare('z', 'sv')                               // 1// in German, ä has a as the base letter
'ä'.localeCompare('a', 'de', { sensitivity: 'base' })      // 0// in Swedish, ä and a are separate base letters
'ä'.localeCompare('a', 'sv', { sensitivity: 'base' })      // 1
```

cf. https://medium.com/nerd-for-tech/basics-of-javascript-string-localecompare-method-b2aa50207706

Seems these edge cases are not relavant for us

File: src/Frontend/Components/AttributionColumn/__tests__/LicenseField.test.tsx (hunk 2 above)
Comment (benedikt-richter) [reply src/Frontend/Components/AttributionColumn/__tests__/LicenseField.test.tsx:62]: These cases seem not very relevant
